python/tvm/runtime/graph_runtime_factory.py

```diff
@@ -0,0 +1,144 @@
+    graph_runtime_kind: str
+        The kind of graph runtime. Like graphruntime, vm and so on.
+    graph_json_str : str or graph class
+        The graph to be deployed in json format output by nnvm graph.
+        The graph can only contain one operator(tvm_op) that
+        points to the name of PackedFunc in the libmod.
+    libmod : tvm.Module
+        The module of the corresponding function
+    Returns
+    -------
+    graph_module : GraphModule
+        Runtime graph module that can be used to execute the graph.
+    """
+    if not isinstance(graph_json_str, string_types):
+        try:
+            graph_json_str = graph_json_str._tvm_graph_json()
+        except AttributeError:
+            raise ValueError("Type %s is not supported" % type(graph_json_str))
+    fcreate = get_global_func("tvm.graph_runtime_factory.create")
+    args = []
+    for k, v in params.items():
+        args.append(k)
+        args.append(ndarray.array(v))
+    return GraphRuntimeFactoryModule(fcreate(graph_runtime_kind, graph_json_str, libmod, module_name, *args))
+
+
+class GraphRuntimeFactoryModule(Module):
+    """Graph runtime factory module.
+
+    This is a module of graph runtime factory
+
+    Parameters
+    ----------
+    module : Module
+        The interal tvm module that holds the actual graph functions.
+
+    Attributes
+    ----------
+    module : Module
+        The interal tvm module that holds the actual graph functions.
+    """
+
+    def __init__(self, module):
+        self.module = module
+        self._select_module = module["select_module"]
+        self._import_module = module["import_module"]
+        self.selected_module = None
+        self.graph_json = None
+        self.lib = None
+        self.params = {}
+        self.iter_cnt = 0
+        super(GraphRuntimeFactoryModule, self).__init__(self.module.handle)
+
+    def __del__(self):
+        pass
+
+    def runtime_create(self, ctx):
+        """Create the runtime using ctx
+
+        Parameters
+        ----------
+        ctx : TVMContext or list of TVMContext
+        """
+        ctx, num_rpc_ctx, device_type_id = get_device_ctx(self.selected_module, ctx)
+        if num_rpc_ctx == len(ctx):
+            fcreate = ctx[0]._rpc_sess.get_function("tvm.graph_runtime_factory.runtime_create")
+        else:
+            fcreate = get_global_func("tvm.graph_runtime_factory.runtime_create")
+        return fcreate(self.selected_module, *device_type_id)
+
+    def import_module(self, mod):
+        """Create the runtime using ctx
+
+        Parameters
+        ----------
+        mod : GraphRuntimeFactoryModule
+            The graph runtime factory module we want to import
+        """
+        return self._import_module(mod)
+
+    def __getitem__(self, key='default'):
+        """Get specific module
+
+        Parameters
+        ----------
+        key : str
+            The key of module.
+        """
+        self.selected_module = self._select_module(key)
```

Comment:
May be we add some fail safe check with some error messages ?

Reply:
The error check is done inside the C++ side. For example, if you pass one module name doesn't exist, we will raise the exception inside C++ side.